### PR TITLE
Add support for Bazaarvoice's Dynamic Catalog Collection (DCC) feature

### PR DESCRIPTION
**Pull request by jantzenw**

### CODE REVIEW

File: Model/Dcc/CatalogData/CatalogProductBuilder.php [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+    /**
+     * @param \Magento\Catalog\Api\Data\ProductInterface|\Magento\Catalog\Model\Product $product
+     *
+     * @return bool|array
+     */
+    private function getFamilies($product): ?array
+    {
+        if ($this->configProvider->isFamiliesEnabled($product->getStoreId())) {
+            $familyAttributeData['sku'] = $product->getSku();
```
Comment (brobie) [inline Model/Dcc/CatalogData/CatalogProductBuilder.php]: constant?

File: Model/Dcc/CatalogData/CatalogProductBuilder.php [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+     * @return string
+     */
+    private function getProductImageUrl($product, $parentProduct = null)
+    {
+        if ($parentProduct && $parentProduct->getData('small_image')) {
+            $productToUse = $parentProduct;
+        } else {
+            $productToUse = $product;
+        }
+        $imageUrl = $this->imageHelperFactory->create()->init($productToUse, 'product_small_image')->getUrl();
```
Comment (brobie) [inline Model/Dcc/CatalogData/CatalogProductBuilder.php]: constant?

File: Model/Dcc/CatalogData/CatalogProductBuilder.php [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+    }
+
+    /**
+     * @param \Magento\Catalog\Api\Data\ProductInterface|\Magento\Catalog\Model\Product $product
+     *
+     * @return string|null
+     */
+    private function getBrandAttribute($product)
+    {
+        return $this->configProvider->getAttributeCode('brand', $product->getStoreId());
```
Comment (brobie) [inline Model/Dcc/CatalogData/CatalogProductBuilder.php]: constant

File: Api/ConfigProviderInterface.php [hunk 4]
```diff
@@ -0,0 +1,357 @@
+<?php
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Api/ConfigProviderInterface.php]: Fixed.

File: Api/ConfigProviderInterface.php [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+     * @param int    $storeId
+     * @param string $scope
+     *
+     * @return bool
+     */
+    public function canSendPurchaseFeed($storeId, $scope = ScopeInterface::SCOPE_STORE): ?bool;
+
+    /**
+     * @param int|null $storeId
+     *
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Api/ConfigProviderInterface.php]: Fixed.

File: Api/ConfigProviderInterface.php [hunk 6]
```diff
@@ -0,0 +1,357 @@
...
+     * @param string   $scope
+     *
+     * @return bool|null
+     */
+    public function isQaEnabled($storeId = null, $scope = ScopeInterface::SCOPE_STORE): ?bool;
+
+    /**
+     * @param int|null $storeId
+     *
```
[the remark this answers is not in the document]
Comment (jantzenw) [reply Api/ConfigProviderInterface.php]: Fixed.

File: Model/Dcc/CatalogData/CatalogProductBuilder.php (hunk 1 above)
Comment (jantzenw) [reply Model/Dcc/CatalogData/CatalogProductBuilder.php]: Agreed. Fixed.

File: Model/Dcc/CatalogData/CatalogProductBuilder.php (hunk 2 above)
Comment (jantzenw) [reply Model/Dcc/CatalogData/CatalogProductBuilder.php]: Agreed. Fixed.

File: Model/Dcc/CatalogData/CatalogProductBuilder.php (hunk 3 above)
Comment (jantzenw) [reply Model/Dcc/CatalogData/CatalogProductBuilder.php]: Fixed.

**Pull request merged**